Hazel/src/Hazel/Core/Base.h

```diff
@@ -2,76 +2,36 @@
 
 #include <memory>
 
-// Platform detection using predefined macros
-#ifdef _WIN32
-	/* Windows x64/x86 */
-	#ifdef _WIN64
-		/* Windows x64  */
-		#define HZ_PLATFORM_WINDOWS
-	#else
-		/* Windows x86 */
-		#error "x86 Builds are not supported!"
-	#endif
-#elif defined(__APPLE__) || defined(__MACH__)
-	#include <TargetConditionals.h>
-	/* TARGET_OS_MAC exists on all the platforms
-	 * so we must check all of them (in this order)
-	 * to ensure that we're running on MAC
-	 * and not some other Apple platform */
-	#if TARGET_IPHONE_SIMULATOR == 1
-		#error "IOS simulator is not supported!"
-	#elif TARGET_OS_IPHONE == 1
-		#define HZ_PLATFORM_IOS
-		#error "IOS is not supported!"
-	#elif TARGET_OS_MAC == 1
-		#define HZ_PLATFORM_MACOS
-		#error "MacOS is not supported!"
-	#else
-		#error "Unknown Apple platform!"
-	#endif
-/* We also have to check __ANDROID__ before __linux__
- * since android is based on the linux kernel
- * it has __linux__ defined */
-#elif defined(__ANDROID__)
-	#define HZ_PLATFORM_ANDROID
-	#error "Android is not supported!"
-#elif defined(__linux__)
-	#define HZ_PLATFORM_LINUX
-	#error "Linux is not supported!"
-#else
-	/* Unknown compiler/platform */
-	#error "Unknown platform!"
-#endif // End of platform detection
+#include "Hazel/Core/PlatformDetection.h"
 
 #ifdef HZ_DEBUG
-	#if defined(HZ_PLATFORM_WINDOWS)
-		#define HZ_DEBUGBREAK() __debugbreak()
-	#elif defined(HZ_PLATFORM_LINUX)
-		#include <signal.h>
-		#define HZ_DEBUGBREAK() raise(SIGTRAP)
-	#else
-		#error "Platform doesn't support debugbreak yet!"
-	#endif
-	#define HZ_ENABLE_ASSERTS
+#if defined(HZ_PLATFORM_WINDOWS)
+#define HZ_DEBUGBREAK() __debugbreak()
+#elif defined(HZ_PLATFORM_LINUX)
+#include <signal.h>
+#define HZ_DEBUGBREAK() raise(SIGTRAP)
+#else
+#error "Platform doesn't support debugbreak yet!"
+#endif
+#define HZ_ENABLE_ASSERTS
 #else
-	#define HZ_DEBUGBREAK()
+#define HZ_DEBUGBREAK()
 #endif
 
 // TODO: Make this macro able to take in no arguments except condition
 #ifdef HZ_ENABLE_ASSERTS
-	#define HZ_ASSERT(x, ...) { if(!(x)) { HZ_ERROR("Assertion Failed: {0}", __VA_ARGS__); HZ_DEBUGBREAK(); } }
-	#define HZ_CORE_ASSERT(x, ...) { if(!(x)) { HZ_CORE_ERROR("Assertion Failed: {0}", __VA_ARGS__); HZ_DEBUGBREAK(); } }
+#define HZ_ASSERT(x, ...) { if(!(x)) { HZ_ERROR("Assertion Failed: {0}", __VA_ARGS__); HZ_DEBUGBREAK(); } }
+#define HZ_CORE_ASSERT(x, ...) { if(!(x)) { HZ_CORE_ERROR("Assertion Failed: {0}", __VA_ARGS__); HZ_DEBUGBREAK(); } }
 #else
-	#define HZ_ASSERT(x, ...)
-	#define HZ_CORE_ASSERT(x, ...)
+#define HZ_ASSERT(x, ...)
+#define HZ_CORE_ASSERT(x, ...)
 #endif
 
 #define BIT(x) (1 << x)
 
 #define HZ_BIND_EVENT_FN(fn) [this](auto&&... args) -> decltype(auto) { return this->fn(std::forward<decltype(args)>(args)...); }
 
 namespace Hazel {
-
 	template<typename T>
```

Comment:
We always leave an empty line after the Hazel namespace
```suggestion
namespace Hazel {

	template<typename T>
```